core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -162,8 +161,6 @@ public DefaultTopologyMonitor(InternalDriverContext context) {
     DriverChannel channel = controlConnection.channel();
     EndPoint localEndPoint = channel.getEndPoint();
 
-    savePort(channel);
```

Comment:
I removed this altogether because I think it is racy.

Reply:
Sorry what I meant by racy is the fact that the control node can change while you are still updating the nodes after having queried the previous control node. It's the same caveat that is already expressed a few lines below in the javadocs of `nodeInfoBuilder`: "it would be racy to call {@code controlConnection.channel().getEndPoint()} from within this method, as the control connection may have changed its channel since."

Reply:
I don't follow either.
We retrieved the channel two lines above, and we reuse that local variable. It would be racy if we called `controlConnection.channel()` again.
Besides, this value is only used for Cassandra < 4, where the port is the same for all nodes.

Reply:
The previous code would only save the port if its current value is -1. So at least in theory, we could save the port of channel A, then change the control channel, and we would keep using port from channel A to create `NodeInfo` instances retrieved from channel B. Granted, port numbers are meant to be the same across nodes with C* < 4, but I do believe that the new version is conceptually more accurate and future-proof than the previous one. And in fact, we really don't need to store this value as an instance field, it is easily obtainable from the current channel so that is another argument in favor of getting rid of this field alttogether.

Reply:
Yes, that's the intended behavior: the ports are all the same, so we save it for the first channel, then reuse it.

> more accurate and future-proof

The only way the current code could fail is if nodes had different ports, but we couldn't find that information in system tables. In that situation, using the port of the control node for another node would always be wrong, whether it's the first ever control node or the current control node.

Please, let's focus this PR on JAVA-2303 and nothing else.